pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1270,6 +1270,11 @@ public void addTable(TableConfig tableConfig)
         if (getAllTables().contains(tableNameWithType)) {
           throw new TableAlreadyExistsException("Table " + tableNameWithType + " already exists");
         }
+        List<SegmentZKMetadata> segmentsZKMetadata = getSegmentsZKMetadata(tableNameWithType);
```

Comment:
No need to actually read the `SegmentZKMetadata`, checking if the dir exists should be enough

Reply:
should we at least list out the dir content? (would there ever be an empty dir?)

Reply:
If you list things out, please limit the list size to (say) 5 segments. You can indicate in the message that there are more (e.g. `...` could be used).